### PR TITLE
feat: introduce `pipeline` crate

### PR DESCRIPTION
**Pull request by shuiyisong**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This is a pre-pr for https://github.com/GreptimeTeam/greptimedb/pull/4014 to introduce https://github.com/GreptimeTeam/pipeline into `src/pipeline/src/etl`. The code is not referred anywhere now.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Comment by tisonkun** [comment]
This is of course a new feature rather than "chore" :D

**Review by waynexia — APPROVED** [review]
The original pipeline repo has a detailed pr-review history. Just merge this mv operation

**Review by killme2008 — APPROVED** [review]
It's a huge PR.  But let's move forward.

File: src/pipeline/src/etl/processor/cmcd.rs [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+use urlencoding::decode;
+
+use crate::etl::field::{Field, Fields};
+use crate::etl::processor::{
+    yaml_bool, yaml_field, yaml_fields, FIELDS_NAME, FIELD_NAME, IGNORE_MISSING_NAME,
+};
+use crate::etl::value::{Map, Value};
+
+pub(crate) const PROCESSOR_CMCD: &str = "cmcd";
```
Comment (tisonkun) [inline src/pipeline/src/etl/processor/cmcd.rs:23]: What does "cmcd" stand for?

Comment (tisonkun) [reply src/pipeline/src/etl/processor/cmcd.rs:23]: I found "Common Media Client Data" below now :D

**Pull request merged**